### PR TITLE
Unification navigation, less bundle size

### PR DESCRIPTION
**Pull request by mikekubn**
-  Adjustmented navigation, for secondary navigation dropdown.
-  Used clx for  less bundle size. Unification mobile and desktop header with navigation. 
- Suggestion is according tailwind guidelines.
- related [PR - https://github.com/cesko-digital/movapp/pull/171](https://github.com/cesko-digital/movapp/pull/171)
- [x] Split component - Header / Navigation

### CODE REVIEW

File: components/basecomponents/Navigation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        active: true,
+      };
+    }
+
+    return { ...item };
+  });
+
+  return (
+    <nav className="w-full pr-10">
+      <div className="container flex flex-wrap justify-end items-center mx-auto" ref={ref}>
```
Comment (JuroUhlar) [inline components/basecomponents/Navigation/index.tsx:48]: 1. ref should be named more specifically `clickOutsideRef` or something
2. it should not be attached here to the menu container, but to the dropdown container. Right now you have a bug where clicking on a menu item does not close the dropdown. 
3. The click outside logic should be handled inside the Dropdown component entirely.

File: components/basecomponents/Navigation/Dropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    <div className="md:absolute md:top-12">
+      <ul className={clsx({ [styles.basic]: true, hidden: !visible || !dropdown.open })}>
+        {items.map(({ name, link, active }, index) => {
+          return (
+            <li
+              key={index}
+              id={id}
+              className={clsx({
+                'text-primary-yellow md:text-gray-700 md:list-disc': active,
+                'hover:text-primary-blue my-2 md:my-1 text-sm': true,
```
Comment (JuroUhlar) [inline components/basecomponents/Navigation/Dropdown/index.tsx:51]: The links inside dropdown should be clickable in their entire area, not just their text :)

**Review by JuroUhlar — CHANGES_REQUESTED** [review]
I have commented a few suggestions you can implement if you want. But I think making our own dropdown component from scratch is a mistake a this is a good opportunity to adopt a headless third party component instead: https://www.radix-ui.com/docs/primitives/components/dropdown-menu

It will solve all of the problems we have now and also most of the future problems that will come up later (accesibility, positioning on mobile, keyboard control etc).

File: components/basecomponents/Navigation/Dropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (mikekubn) [reply components/basecomponents/Navigation/Dropdown/index.tsx:1]: It's depends on project folder structure. I will change this according suggestions.

File: components/basecomponents/Navigation/Dropdown/index.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import { OpenDropdown } from '..';
+
+interface DropdownItems extends SubmenuItem {
+  active?: boolean;
+}
+
+interface DropdownProps {
+  submenu: SubmenuItem[];
+  dropdown: OpenDropdown;
+  id: string;
```
[the remark this answers is not in the document]
Comment (mikekubn) [reply components/basecomponents/Navigation/Dropdown/index.tsx:17]: 24: `const visible = dropdown.id === id`

File: components/basecomponents/Navigation/index.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  active?: boolean;
+}
+
+export const Navigation: React.FC<{ showNavigation: boolean }> = ({ showNavigation }) => {
+  const { asPath } = useRouter();
+  const { t } = useTranslation('common');
+  const [dropdown, setDropdown] = React.useState<OpenDropdown>({ open: false });
+  const { ref } = useClickOutside<HTMLDivElement>(() => setDropdown({ open: false }));
+
+  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
```
[the remark this answers is not in the document]
Comment (mikekubn) [reply components/basecomponents/Navigation/index.tsx:26]: It can be.

File: components/basecomponents/Navigation/index.tsx (hunk 1 above)
Comment (mikekubn) [reply components/basecomponents/Navigation/index.tsx:48]: Ref should be named more specifically if you have more than one ref in component.

**Comment by mikekubn** [comment]
> I have commented a few suggestions you can implement if you want. But I think making our own dropdown component from scratch is a mistake a this is a good opportunity to adopt a headless third party component instead: https://www.radix-ui.com/docs/primitives/components/dropdown-menu
> 
> It will solve all of the problems we have now and also most of the future problems that will come up later (accesibility, positioning on mobile, keyboard control etc).

@JuroUhlar it's make sense. I will close ticket

**Pull request closed without merging**